### ISSUE TITLE
Define canonical IMAS radiation mapping for reconstructed emissivity and tomography products

### ISSUE THREAD

**Issue by HongSik-Yun-Fusion** [VEST-Tokamak/vaft#1156]
## Background

VAFT already has several planned or implemented workflows involving radiation diagnostics and tomographic reconstruction:

- #267 — classical SXR tomography with general 2-D, flux-aligned, and iso-flux reconstruction;
- #886 — equilibrium-aware magnetic-island emissivity phantoms and synthetic SXR forward modeling;
- #991 — VAFT-native PyTorch rotational SXR tomography;
- #1001 — multi-array / multi-view rotational SXR tomography;
- #669 — common ML training, inference, provenance, and `vaft-nn` model-artifact distribution;
- #909 — solver-neutral mesh infrastructure with future IMAS `grid_ggd` support;
- #1127–#1132 — IMAS-native `DD` / occurrence / storage architecture.

These issues define the measurement, forward-model, inversion, ML, mesh, and persistence layers well, but one architectural question remains unresolved:

> **What is the canonical IMAS representation of a reconstructed radiation field such as**
>
> \[
> \epsilon(R,Z,t),
> \]
>
> **independently of whether it was reconstructed by classical tomography, a neural network, or another inverse method?**

At present, VAFT can naturally represent the measured SXR signals in `soft_x_rays`, but the tomography issues do not define a common persisted IMAS output for the reconstructed emissivity field.

This issue should establish that contract.

---

## Core principle

VAFT should distinguish three scientific layers:

```text
diagnostic measurement
        ↓
physical reconstructed field
        ↓
physical interpretation
```

For radiation diagnostics:

```text
soft_x_rays / bolometer / other radiation diagnostics
        ↓
tomographic or learned reconstruction
        ↓
radiation
        ↓
MHD / island / transport / radiation analysis
```

The corresponding responsibilities should be:

### Diagnostic IDS

Represents what an instrument measured.

Examples:

```text
soft_x_rays
bolometer
```

This layer owns quantities such as:

- detector and aperture geometry;
- line of sight;
- spectral / filter response;
- measured brightness or power;
- channel time bases;
- acquisition metadata.

### `radiation` IDS

Represents the inferred physical radiation field.

Examples:

\[
\epsilon(R,Z,t),
\]

and, where scientifically justified in future workflows,

\[
\epsilon(R,Z,\phi,t).
\]

This layer should be independent of the reconstruction method.

### MHD / topology representation

Represents the magnetic structure used to interpret or constrain the radiation field.

Examples:

- rational surface;
- magnetic island;
- \(m/n\);
- island width;
- island phase;
- perturbed magnetic fields;
- O-point / X-point topology.

A magnetic island and an emissivity perturbation are not the same quantity and must not be conflated.

---

# Goal

Define a machine-independent VAFT contract in which reconstructed radiation fields are represented canonically through IMAS `radiation`, while retaining the original diagnostic IDS as the measurement source.

The same contract should support results from:

```text
classical tomography
regularized tomography
flux-aligned tomography
iso-flux reconstruction
rotational tomography
machine-learning tomography
future Bayesian reconstruction
```

without introducing method-specific physical schemas.

---

# 1. Measurement and reconstruction must remain separate

A typical SXR workflow should become:

```text
soft_x_rays
    measured LOS brightness
            │
            │ reconstruction
            ▼
radiation
    reconstructed emissivity
```

For example:

```text
soft_x_rays
    channel geometry
    spectral response
    brightness(t)
        ↓
#267 classical tomography
        ↓
radiation
    emissivity(R,Z,t)
```

and:

```text
soft_x_rays
        ↓
#991 learned inverse model
        ↓
radiation
    emissivity(R,Z,t)
```

The reconstruction backend must not change the scientific meaning of the output.

---

# 2. `radiation` should be the canonical physical output

The baseline physical result should be a radiation emissivity field with explicit:

- spatial coordinates / grid;
- time;
- emissivity values;
- units or relative-unit semantics;
- validity information where appropriate;
- code / reconstruction provenance;
- references to source diagnostic data.

Exact IMAS Data Dictionary paths must be resolved and validated against the supported DD version during implementation rather than duplicated as VAFT-specific schema.

Conceptually:

```text
radiation
├── grid / grid_ggd
├── reconstructed emissivity
├── time
├── code
└── provenance
```

VAFT should not create a parallel persistent structure such as:

```text
sxr_tomography.emissivity
ml_tomography.image
island_tomography.result
```

when the quantity being represented is physically the same radiation field.

---

# 3. Reconstruction method belongs to provenance, not field semantics

The following should all be allowed to produce the same physical output contract:

```text
Tikhonov
MFR
SVD
NNLS
Bayesian inversion
PyTorch model
future surrogate
```

Method identity belongs in provenance.

For example:

```text
radiation:sxr-tomography
    physical quantity:
        emissivity field

    provenance:
        method = Tikhonov
        regularization parameters
        VAFT revision
        source soft_x_rays instance
```

or:

```text
radiation:sxr-tomography-ml
    physical quantity:
        emissivity field

    provenance:
        model identity
        model version / digest
        training dataset identity
        VAFT revision
        source soft_x_rays instance
```

Changing a solver parameter, Git commit, or model checkpoint must not automatically redefine the physical quantity itself.

---

# 4. ML model artifacts remain outside the IDS payload

This issue must preserve the separation established by #669.

A trained neural-network model is a reusable computational artifact, not a shot-level radiation field.

Therefore:

```text
IMAS / DD
    radiation
        reconstructed physical result

vaft-nn
    ModelArtifact
        architecture
        weights
        manifest
        model card
        training provenance
        immutable version
```

The `radiation` provenance should reference the model artifact when a learned inverse model was used.

Do not place PyTorch / ONNX / safetensors model weights directly into the `radiation` IDS.

---

# 5. Use one output contract for #267 and #991

#267 and #991 currently approach the same physical inverse problem through different methods.

They should converge on one reconstruction result abstraction.

Conceptually:

```python
result = reconstruct_radiation(...)
```

with scientific content equivalent to:

```text
grid
emissivity
time
uncertainty / quality information
provenance
```

Possible backends may include:

```python
method="tikhonov"
method="mfr"
method="nnls"
method="neural_network"
```

The exact public API belongs to the tomography/process architecture and need not be frozen by this issue, but all supported backends should be exportable to the same IMAS `radiation` representation.

---

# 6. Regular R-Z reconstruction is the baseline

The first supported representation should cover the current #267 / #991 target:

\[
\epsilon(R,Z,t).
\]

A regular structured \(R,Z\) reconstruction must not depend on the completion of the generic unstructured-mesh infrastructure.

Conceptually:

```text
#267 / #991
      ↓
regular R-Z radiation field
      ↓
IMAS radiation
```

This is the minimum implementation required by this issue.

---

# 7. Preserve compatibility with `grid_ggd`

The representation contract should not assume permanently that tomography is a rectangular image.

Future reconstructed fields may live on:

- regular \(R,Z\) grids;
- triangular / unstructured poloidal meshes;
- flux-aligned meshes;
- island-aware meshes;
- solver-generated GGD meshes;
- eventually 3-D \(R,Z,\phi\) meshes.

Therefore the output abstraction should remain compatible with IMAS GGD where appropriate.

Relationship to #909:

```text
baseline tomography
    regular R-Z grid
    independent of #909

advanced tomography
    unstructured / topology-aware mesh
              ↓
             #909
              ↓
           grid_ggd
```

#909 must therefore be an extension dependency, not a blocker for basic SXR tomography.

---

# 8. Magnetic-island topology must remain distinct from radiation emissivity

#886 already establishes the correct separation:

```text
magnetic-island topology
        ↓
emissivity response model
        ↓
SXR forward projection
```

This issue should extend that separation into the persisted scientific representation:

```text
magnetic topology
    equilibrium / MHD representation
             │
             │ physical relation
             ▼
radiation
    emissivity field
             │
             │ forward diagnostic response
             ▼
soft_x_rays
    LOS brightness
```

For an experimentally reconstructed island-like emissivity structure:

```text
soft_x_rays
        ↓
tomography
        ↓
radiation ε(R,Z,t)
        ↓
comparison / interpretation
        ↓
MHD island topology
```

Do not encode quantities such as:

```text
m
n
island width
rational-surface identity
magnetic perturbation amplitude
```

as if they were intrinsic radiation-field coordinates unless the IMAS schema explicitly assigns that meaning.

---

# 9. Island-shaped emissivity does not require an island-shaped grid

A reconstructed magnetic-island signature can initially be represented simply as:

\[
\epsilon(R,Z,t)
\]

on an ordinary poloidal mesh.

The island morphology is then contained in the scalar field itself.

Therefore:

```text
island tomography
    ≠
mandatory island-topology mesh
```

A topology-aware mesh may later improve analysis or numerical efficiency, but it is not required to represent an island-shaped radiation field.

This distinction prevents #267 from becoming unnecessarily dependent on advanced mesh work.

---

# 10. Allow future topology-aware / island-aligned representation

After #909 and related coordinate infrastructure mature, VAFT may support radiation fields on meshes aligned with:

- equilibrium flux surfaces;
- rational surfaces;
- island separatrices;
- helical coordinates;
- O-point / X-point structure.

A possible future workflow is:

```text
MHD / equilibrium
    island-aware GGD
          │
          ├── magnetic quantities
          │
          └── radiation emissivity
```

or equivalent cross-IDS grid referencing where supported by the target IMAS Data Dictionary.

This issue should define compatibility with that direction without requiring its implementation now.

---

# 11. Keep 3-D tomography out of the initial implementation

#267 correctly treats 3-D tomography as out of scope.

The current required target remains:

\[
\epsilon(R,Z,t).
\]

Future multi-view / toroidally distributed diagnostics may motivate:

\[
\epsilon(R,Z,\phi,t).
\]

That work may eventually depend on:

- #1001 multi-array geometry;
- #909 generic GGD / mesh infrastructure;
- #1108 3-D equilibrium interoperability;
- future 3-D magnetic-coordinate infrastructure.

This issue should ensure that the data contract does not prohibit 3-D radiation fields, but no 3-D inversion algorithm is required.

---

# 12. Integrate with the occurrence-native DD architecture

The new representation must fit #1127–#1132 rather than introduce a tomography-specific storage hierarchy.

Multiple scientifically distinct reconstructed radiation products may coexist as semantic IDS instances.

Conceptually, a deployment may eventually define entries such as:

```text
radiation:
    sxr-tomography
    sxr-tomography-ml
```

The exact VEST occurrence assignments belong to #1129 and the deployment registry.

The user-facing scientific identity should remain semantic:

```python
dd.view(radiation="sxr-tomography")
```

rather than exposing occurrence integers or storage paths.

---

# 13. Do not create occurrences for implementation versions

Follow the identity principles of #1128.

The following should normally remain provenance:

```text
regularization parameter
solver tolerance
VAFT commit
PyTorch version
model SHA
training seed
model checkpoint version
```

They should not automatically produce persistent semantic instances such as:

```text
sxr-tomography-v17
sxr-tomography-lambda-0.01
sxr-nn-checkpoint-42
```

A new semantic instance is justified only when the scientific interpretation or intended product identity differs.

For example, the distinction between:

```text
sxr-tomography
sxr-tomography-ml
```

may be useful if the deployment wants both as persistent independently selectable scientific products.

The final choice belongs to the deployment registry.

---

# 14. Provenance must preserve source relationships

A persisted reconstructed `radiation` product should be traceable to at least:

```text
source diagnostic instance
reconstruction code / method
input equilibrium if used
geometry identity
parameters
software revision
external model artifact where applicable
```

For flux-aligned tomography:

```text
soft_x_rays
equilibrium
     ↓
reconstruction
     ↓
radiation
```

For learned rotational tomography:

```text
soft_x_rays
equilibrium / geometry as required
ModelArtifact
       ↓
inference
       ↓
radiation
```

These relationships should use the general VAFT/IMAS provenance and registry mechanisms rather than method-specific ad-hoc metadata.

---

# 15. Bolometer and other radiation diagnostics should remain compatible

The contract should be defined around the physical field rather than SXR specifically.

Conceptually:

```text
soft_x_rays ──┐
              │
bolometer ─────┼──► reconstruction ──► radiation
              │
future diag ──┘
```

Diagnostic-specific spectral-response semantics remain with the source diagnostic.

The reconstructed `radiation` product should retain enough provenance to distinguish what radiation quantity was actually inferred.

Do not assume that all radiation diagnostics observe the same spectral range or physical emissivity.

---

# Proposed implementation phases

## Phase 1 — Data-model review

- inspect the supported IMAS Data Dictionary version;
- confirm the current `radiation` field / GGD semantics;
- identify the exact nodes suitable for reconstructed emissivity;
- determine how source relationships and code provenance should be represented;
- document any DD limitation explicitly.

## Phase 2 — VAFT reconstruction result contract

Define a backend-neutral radiation reconstruction result sufficient for:

- #267 classical tomography;
- #991 ML tomography.

At minimum preserve:

```text
spatial grid
emissivity
time
units / normalization semantics
quality / uncertainty metadata where available
provenance
```

## Phase 3 — IMAS mapping

Implement conversion:

```text
VAFT reconstructed radiation result
        ↕
IMAS radiation IDS
```

with round-trip validation where practical.

## Phase 4 — Classical tomography integration

Make #267 results exportable to the canonical `radiation` representation.

## Phase 5 — ML tomography integration

Make #991 inference results exportable to the same `radiation` representation while keeping `ModelArtifact` ownership in #669 / `vaft-nn`.

## Phase 6 — DD / registry integration

Integrate persisted products with #1127–#1132 and define any VEST-specific semantic instances through #1129.

## Phase 7 — GGD extension

After #909 provides the necessary infrastructure, validate at least one non-rectangular or unstructured reconstructed radiation field.

This phase may be completed in a follow-up issue if the baseline contract is already stable.

---

# Acceptance criteria

- [ ] Review the target IMAS `radiation` IDS and document the exact supported mapping.
- [ ] Establish `radiation` as the canonical IMAS physical representation for reconstructed radiation emissivity fields.
- [ ] Keep `soft_x_rays`, `bolometer`, and other diagnostic IDSs as measurement-layer representations.
- [ ] Define one backend-neutral VAFT reconstruction result contract.
- [ ] Support baseline \(\epsilon(R,Z,t)\) mapping.
- [ ] Preserve explicit spatial coordinates, time, units / normalization semantics, and provenance.
- [ ] Record source diagnostic relationships.
- [ ] Record equilibrium / geometry dependencies when reconstruction uses them.
- [ ] Make #267 classical tomography output compatible with the contract.
- [ ] Make #991 ML tomography output compatible with the same contract.
- [ ] Keep NN weights and model artifacts under #669 / `vaft-nn`, not inside shot-level radiation data.
- [ ] Ensure reconstruction method and implementation version are provenance rather than physical field semantics.
- [ ] Keep magnetic-island topology distinct from radiation emissivity.
- [ ] Do not require an island-specific mesh for island-shaped emissivity.
- [ ] Ensure the contract is compatible with future `grid_ggd` / unstructured representations.
- [ ] Ensure the baseline implementation does not depend on completion of #909.
- [ ] Integrate persistent products with the occurrence-native DD / registry architecture.
- [ ] Do not allocate occurrences merely for solver parameters or model versions.
- [ ] Add tests covering classical and ML results mapping to the same physical schema.
- [ ] Add at least one round-trip or schema-validation example using a reconstructed radiation field.

---

# Out of scope

This issue does not require:

- implementing the inversion algorithms owned by #267;
- implementing the magnetic-island model owned by #886;
- training the neural model owned by #991;
- multi-array extension owned by #1001;
- generic ML artifact infrastructure owned by #669;
- completion of the full mesh infrastructure in #909;
- 3-D tomography;
- a self-consistent nonlinear MHD island reconstruction;
- atomic-radiation spectral inversion;
- joint \(T_e\), impurity, and emissivity inference;
- storing neural-network weights inside IMAS;
- defining universal occurrence numbers across deployments.

---

# Relationship to existing issues

### Direct consumers

- #267 — classical SXR tomography should publish this physical representation.
- #991 — learned SXR tomography should publish the same physical representation.
- #1001 — future multi-array reconstruction should retain the same contract.

### Upstream physical model

- #886 — defines the separation between magnetic topology, emissivity response, and synthetic diagnostic observation.

### ML artifact layer

- #669 — owns model training, versioning, model artifacts, and `vaft-nn`.

### Geometry / mesh extension

- #909 — provides future solver-neutral unstructured meshes and `grid_ggd` infrastructure.

### 3-D extension

- #1108 — relevant when reconstruction becomes genuinely non-axisymmetric / 3-D.

### Data architecture

- #1127 — `DD` / `DDView` scientific aggregate model.
- #1128 — semantic IDS instances and occurrence-native architecture.
- #1129 — VEST deployment occurrence registry.
- #1132 — native IMAS backing and occurrence-aware storage.

### Existing measurement publication

- #599 — established canonical publication of measured `soft_x_rays`.

---

# Target architecture

```text
                         ┌───────────────┐
                         │   vaft-nn     │
                         │ ModelArtifact │
                         └───────┬───────┘
                                 │
                                 │ provenance
                                 ▼
soft_x_rays ───────┐      ┌───────────────┐
                   ├─────►│ reconstruction│
bolometer ─────────┤      │ classical / ML│
                   │      └───────┬───────┘
future diagnostics ┘              │
                                  ▼
                          ┌───────────────┐
                          │   radiation   │
                          │ ε(R,Z,t)      │
                          │ future GGD    │
                          └───────┬───────┘
                                  │
                    ┌─────────────┴─────────────┐
                    ▼                           ▼
                analysis                  MHD/topology
                                           comparison
```

The intended architectural rule is:

> **Diagnostic IDSs store what was measured; `radiation` stores the reconstructed physical radiation field; reconstruction methods and trained models are recorded through provenance and external artifacts; magnetic topology remains a separate physical object.**
